ui/src/loudml/utils/model.js

```diff
@@ -1,22 +1,53 @@
 import moment from 'moment'
 
+const MIN_INTERVAL_SECOND = 5
+const MIN_INTERVAL_UNIT = `${MIN_INTERVAL_SECOND}s`
+const MIN_SPAN = 10
+
 export const normalizeInterval = bucketInterval => {
+    // interval = max(5, min(bucketIntervak, 60))
     const regex = /(\d+)(.*)/
     const interval = regex.exec(bucketInterval)
+    if (!interval) {
+        return MIN_INTERVAL_UNIT
+    }
+
     const duration = moment.duration(Number.parseInt(interval[1], 10), interval[2]).asSeconds()
+    if (!duration) {
+        return MIN_INTERVAL_UNIT
+    }
+
     const normalized = Math.max(
-        5,
-        moment.duration(
-            Math.min(
-                duration,
-                60
-            ),
-            's'
-        ).asSeconds()
+        MIN_INTERVAL_SECOND,
+        Math.min(
+            duration,
+            60
+        )
     )
     return `${normalized}s`
 }
 
+export const normalizeSpan = bucketInterval => {
+    // span = min(10, (2h/bucketInterval))
```

Comment:
Sam, based on a few tests, we can select a value between 0 and 100:
span = max(
    10,
    min(24h/bucket_interval, 100)
)

Reply:
in next commit